### PR TITLE
OSSM 3309: Add maistra istio operator push image jobs

### PR DESCRIPTION
**Pull request by yxun**
Signed-off-by: Yuanlin <yuanlin.xu@redhat.com>

### CODE REVIEW

**Comment by yxun** [comment]
pj-rehearse

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.4.yaml [hunk 1]
```diff
@@ -60,6 +65,67 @@ tests:
...
+          memory: 100Mi
+      timeout: 10m0s
+    - as: istio-operator-push-image-2-4
+      cli: latest
+      commands: |
+        export QUAY_PASS=$(cat /usr/local/ci-secrets/maistra-dev/password)
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          entrypoint \
+          sh -c \
+          "docker login -u='maistra-dev+prow' \
```
[the remark this answers is not in the document]
Comment (yxun) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.4.yaml:107]: - I can try to use `/maistra-secrets` base dir.

- The docker login command is executed in the `oc rsh` shell that means your suggestion will not work. There is no mounted file in the builder container.
The mounted secret file is in the prow job container and then it triggers an oc rsh command to run docker commands in a remote cluster maistra builder container.

That's why I cat the secret before running oc rsh command. From the doc I read, prow will automatically mask any secret content in logs even if the job cat the file.

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.4.yaml [hunk 2]
```diff
@@ -60,6 +65,67 @@ tests:
...
+          -p=${QUAY_PASS} quay.io; \
+          make push \
+          IMAGE=quay.io/maistra-dev/istio-ubi8-operator:PR-${PULL_NUMBER}"
+      credentials:
+      - mount_path: /usr/local/ci-secrets/maistra-dev
+        name: maistra-dev-prow
+        namespace: test-credentials
```
[the remark this answers is not in the document]
Comment (yxun) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.4.yaml:114]: The credential was set in their Vault server :
https://vault.ci.openshift.org/ui/vault/auth?with=oidc%2F
https://docs.ci.openshift.org/docs/how-tos/adding-a-new-secret-to-ci/

Then then their CI automatically sync it into a the prow cluster  namespace: test-credentials every 30 mins

[the remark this answers is not in the document]
Comment (jwendell) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.4.yaml:114]: Thanks!

**Pull request merged**